WPMedia/fusion-news-theme-blocks
# TMEDIA-253: Change all promos to render null if there is no content instead of a …

## Description
Do not render promos if there is no content

## Jira Ticket
- [TMEDIA-253](https://arcpublishing.atlassian.net/browse/TMEDIA-253)

## Acceptance Criteria
If no content is present for one of these promo blocks, then nothing should render on the page (including the issues described in this ticket: no fallback promo image, nor spacing around it)

## Test Steps
1. Checkout this branch `git checkout TMEDIA-253-do-not-render-no-content-promos`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-manual-promo-block,@wpmedia/extra-large-promo-block,@wpmedia/large-manual-promo-block,@wpmedia/large-promo-block,@wpmedia/medium-manual-promo-block,@wpmedia/medium-promo-block,@wpmedia/shared-styles,@wpmedia/shared-styles,@wpmedia/shared-styles,@wpmedia/small-manual-promo-block,@wpmedia/small-promo-block`
3. Load up http://localhost/pagebuilder/editor/curate?p=pPdo2UO0eFD57Prs&v=vhpcVNxjdFD57Prs
4. Note the space above the promotions on the right bar:
![image](https://user-images.githubusercontent.com/2287238/122296561-8e3bd900-cec8-11eb-9854-5331629e5c96.png)
5. Add a promo item with no/invalid content specified:
![image](https://user-images.githubusercontent.com/2287238/122296714-b592a600-cec8-11eb-870c-cd1955d6da29.png)
![image](https://user-images.githubusercontent.com/2287238/122296753-c0e5d180-cec8-11eb-911a-95c0e382153f.png)
6. Verify that it does not change the layout:
![image](https://user-images.githubusercontent.com/2287238/122296811-d6f39200-cec8-11eb-806c-857e21b35379.png)
7. Repeat for extra-large, medium, and small

## Effect Of Changes
### Before
When adding a promo with no content or invalid content, the page would include the image with a replacement button:
![image](https://user-images.githubusercontent.com/2287238/122297260-53867080-cec9-11eb-9e6b-308e9f6dcdad.png)

### After
No content will show up unless properly configured:
![image](https://user-images.githubusercontent.com/2287238/122297423-7e70c480-cec9-11eb-9fdd-c2545f5b5523.png)

## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
